utils.go

```diff
@@ -722,22 +783,47 @@ func checkMntSrcOwnership(mntSrc string, uid, gid uint32) (bool, uint32, uint32,
 	mntSrcUid = st.Uid
 	mntSrcGid = st.Gid
 
-	if mntSrcUid != uid || mntSrcGid != gid {
+	// We expect the host uid assigned to the container to be equal or higher
+	// than the uid of the dir being mounted into the container (i.e., the former
+	// is a container-range uid, the latter is a uid in the normal range of host
+	// uids).
+
+	if uid > mntSrcUid && gid > mntSrcGid {
```

Comment:
Why a "&&" and not a "||"? I would expect any of the statements to be enough to require uid-shifting.

Reply:
In the normal case, the container's assigned uid and gid ranges are higher than the uid and gid ranges of the host directory being mounted into `/var/lib/docker`.  

If only uid or gid are higher, it's not a normal case (i.e., it means that the container has uid:gid 165536 but the host directory being mounted has for example uid = 0, gid = 10000000). In this case we want to prevent uid shifting and log a warning as the code that follows does.

We want to do the ID shifting when both the uid and gid needed. If either does not needed there is a likely problem because it means the uid and gid of the file are in very different ranges.

Reply:
In the paragraph above we are saying that the container-uid is >= src-dir-uid, which means that this would be a valid case (container-uid/gid = 165536:165536, src-dir-uid/gid = 165536:0), and as such, ID-shifting should be allowed.

My point is that either we adjust the comment above (container-uid is > src-dir-uid) or we change this "if-statement".

Reply:
Got it; I've reworked the comment to clarify further.